node/rpc/src/lib.rs

```diff
@@ -158,7 +162,52 @@ where
         .with_method("Filecoin.GasEstimateFeeCap", gas_estimate_fee_cap::<DB, KS>)
         .finish_unwrapped();
 
-    let mut app = tide::Server::with_state(rpc);
-    app.at("/rpc/v0").post(handle_json_rpc);
-    app.listen(rpc_endpoint).await.unwrap();
+    let try_socket = TcpListener::bind(rpc_endpoint).await;
+    let listener = try_socket.expect("Failed to bind to addr");
+    let state = Arc::new(rpc);
+    info!("waiting for web socket connections");
+    while let Ok((stream, addr)) = listener.accept().await {
+        task::spawn(handle_connection_and_log(state.clone(), stream, addr));
+    }
+
+    info!("Stopped accepting websocket connections");
+}
+
+async fn handle_connection_and_log(
+    state: Arc<Server<MapRouter>>,
+    tcp_stream: TcpStream,
+    addr: std::net::SocketAddr,
+) {
+    span!("handle_connection", {
+        if let Ok(ws_stream) = async_tungstenite::accept_async(tcp_stream).await {
+            info!("accepted websocket connection at {:}", addr);
+            let (mut ws_sender, ws_receiver) = ws_stream.split();
+            let responses_result = ws_receiver
+                .try_filter(|s| future::ready(!s.is_text()))
+                .try_fold(Vec::new(), |mut responses, s| async {
+                    let request_text = s.into_text()?;
+                    let call: RequestObject = serde_json::from_str(&request_text)
+                        .map_err(|s| Error::Protocol(Cow::from(s.to_string())))?;
+                    let response = state.handle(call).await;
+                    let response_text = serde_json::to_string(&response)
+                        .map_err(|s| Error::Protocol(Cow::from(s.to_string())))?;
+                    responses.push(response_text);
+                    Ok(responses)
+                })
+                .await;
+
+            if let Err(error) = responses_result {
+                error!("error obtaining request {:?}", error)
+            } else {
+                for response_text in responses_result.unwrap() {
+                    ws_sender
+                        .send(Message::text(response_text))
+                        .await
+                        .unwrap_or_else(|s| error!("Error sending response {:?}", s))
+                }
+            }
+        } else {
+            error!("web socket connection failed at {:}", addr)
```

Comment:
is this necessarily a fail? this happens at disconnection too right?

Reply:
That is true, maybe a warn might be better here?

Reply:
Sounds good to me